From a64445c8d335290b61a3f941b0386044a1fc492a Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 26 Aug 2014 14:12:01 -0400
Subject: [ethernet] igb: Convert iounmap to pci_iounmap

Message-id: <1409062359-13063-44-git-send-email-sassmann@redhat.com>
Patchwork-id: 89155
O-Subject: [RHEL7.1 PATCH 43/81] INTEL-IGB: Convert iounmap to pci_iounmap
Bugzilla: 1091121
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Peter Senna Tschudin <peter.senna@gmail.com>

Use pci_iounmap instead of iounmap when the virtual mapping was done
with pci_iomap. A simplified version of the semantic patch that finds this
issue is as follows: (http://coccinelle.lip6.fr/)

// <smpl>
@r@
expression addr;
@@
addr = pci_iomap(...)

@rr@
expression r.addr;
@@
* iounmap(addr)
// </smpl>

Signed-off-by: Peter Senna Tschudin <peter.senna@gmail.com>
Tested-by: Aaron Brown <aaron.f.brown@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 75009b3a88cd8f56315eb4ced296ee2f060cf946)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/igb/igb_main.c b/drivers/net/ethernet/intel/igb/igb_main.c
index f1f4cf4..4e242f2 100644
--- a/drivers/net/ethernet/intel/igb/igb_main.c
+++ b/drivers/net/ethernet/intel/igb/igb_main.c
@@ -2652,7 +2652,7 @@ err_eeprom:
   iounmap(hw->flash_address);
 err_sw_init:
  igb_clear_interrupt_scheme(adapter);
- iounmap(hw->hw_addr);
+ pci_iounmap(pdev, hw->hw_addr);
 err_ioremap:
  free_netdev(netdev);
 err_alloc_etherdev:
@@ -2819,7 +2819,7 @@ static void igb_remove(struct pci_dev *pdev)
  igb_disable_sriov(pdev);
 #endif
 
- iounmap(hw->hw_addr);
+ pci_iounmap(pdev, hw->hw_addr);
  if (hw->flash_address)
   iounmap(hw->flash_address);
  pci_release_selected_regions(pdev,
-- 
1.7.1